charts/opentelemetry-collector/templates/_pod.tpl

```diff
@@ -47,6 +47,10 @@ containers:
           fieldRef:
             fieldPath: spec.nodeName
       {{- end }}
+      {{- if and (.Values.useGOMEMLIMIT) ((((.Values.resources).limits).memory))  }}
+      - name: GOMEMLIMIT
+        value: {{ div (mul (include "opentelemetry-collector.convertMemToMib" .Values.resources.limits.memory) 80) 100 }}Mi
```

Comment:
```suggestion
        value: {{ div (mul (include "opentelemetry-collector.convertMemToMib" .Values.resources.limits.memory) 80) 100 }}MiB
```

`The supported suffixes include B, KiB, MiB, GiB, and TiB.` from the [docs](https://pkg.go.dev/runtime).